lte/gateway/c/session_manager/ShardTracker.cpp

```diff
@@ -13,45 +13,43 @@
 
 #include <string>
 #include <vector>
+#include <set>
 
 #include "ShardTracker.h"
 
 namespace magma {
 
-int ShardTracker::add_ue() {
-  /*
-  * If there are no shards, add an entry for the shard
-  * representing a single UE at the new shard
-  */
-  if (ue_count_per_shard_.size() == 0) {
-    ue_count_per_shard_.push_back(1);
-    return 0;
+uint16_t ShardTracker::add_ue(const std::string imsi) {
+  if (imsis_per_shard_.empty()) {
+    imsis_per_shard_.push_back({imsi});
```

Comment:
the very first shard should be initiated at the constructor. This line will be hit in all the requests and it is only used once.